habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/ContainerizeDepsDockerfileHelper.java

```diff
@@ -58,19 +71,23 @@ public static String createContainerStageContentFrom(String template, String anc
      * @param moduleBaseDir the module base directory
      * @return updated Dockerfile content
      */
-    public static String updateDockerfileWithContainerStageLogic(File dockerFile, String anchorDirectory, String moduleBaseDir) {
-        String builderStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(BUILDER_STAGE_TEMPLATE, anchorDirectory, moduleBaseDir);
-        String finalStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(FINAL_STAGE_TEMPLATE, null, null);
+    public static String updateDockerfileWithContainerStageLogic(File dockerFile, String anchorDirectory, String moduleBaseDir, String owner, String baseImage) {
+        String builderStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(BUILDER_STAGE_TEMPLATE, anchorDirectory, moduleBaseDir, owner, baseImage);
+        String finalStageContent = ContainerizeDepsDockerfileHelper.createContainerStageContentFrom(FINAL_STAGE_TEMPLATE, null, null, owner, baseImage);
         StringBuilder content = new StringBuilder();
-        Boolean builderStageContentIncluded = false;
-        Boolean finalStageContentIncluded = false;
+        boolean builderStageContentIncluded = false;
+        boolean finalStageContentIncluded = false;
+        int firstFromLine = -1;
 
         boolean skipLine = false;
         try (BufferedReader buffer = new BufferedReader(new FileReader(dockerFile))) {
             String line = buffer.readLine();
 
             while (line != null) {
-                line = line.strip();
+                line = line.stripTrailing();
+                if(firstFromLine < 0 && line.strip().startsWith("FROM")) {
```

Comment:
I: `ARG` values that are used in _any_ `FROM` line must be declared before the _first_ `FROM` line.  By always inserting the builder stage logic just prior to the first `FROM`, we ensure that any `ARG` values used in `FROM` are functionally preserved.